### PR TITLE
Fix #5658

### LINKED ISSUE #5658: filterBy: custom (filter object paramter is required)
what is the proper way to implement a custom filter?

https://live.bootstrap-table.com/code/Scnck/7032

### PR DESCRIPTION
**Pull request by fxwiegand**
**🤔Type of Request**
- [x] **Bug fix**
- [ ] **New feature**
- [ ] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**
Fix #5658

**📝Changelog**

- [x] **Core**
- [ ] **Extensions**

Custom filters can now be used without providing dummy objects as described [here](https://github.com/wenzhixin/bootstrap-table/issues/5658#issuecomment-807953782).

**💡Example(s)?**
Before: [Not working](https://live.bootstrap-table.com/code/fxwiegand/11314)

After:  [Working](https://live.bootstrap-table.com/code/fxwiegand/11315)
**☑️Self Check before Merge**

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by djhvscf** [comment]
Please provide an [Online Example](https://live.bootstrap-table.com/) to show this PR working, thanks!

**Comment by fxwiegand** [comment]
Sorry i missed the bottom section of the online example docs where it is explained that `<username>:<branch>`is a valid option.

Before: [Not working](https://live.bootstrap-table.com/code/fxwiegand/11314)

After:  [Working](https://live.bootstrap-table.com/code/fxwiegand/11315)

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1105,7 +1105,6 @@ class BootstrapTable {
       if (this.options.sortReset) {
         this.unsortedData = [...this.data]
       }
-
```
Comment (wenzhixin) [inline src/bootstrap-table.js:1108]: This line does not need to be deleted.

Comment (fxwiegand) [reply src/bootstrap-table.js:1108]: Sorry about that. Was definitely unintentional! I re-added the line.

Comment (fxwiegand) [reply src/bootstrap-table.js:1108]: Sorry about that. Was definitely unintentional! I re-added the line.

**Review by wenzhixin — APPROVED** [review]

**Comment by wenzhixin** [comment]
Thanks for your contribution!

**Pull request merged**